### DISCUSSION TITLE
Retry Topic Auto configuration - Auto creation of topics

### DISCUSSION THREAD

**Discussion by tomazfernandes** [spring-projects/spring-kafka#2652] [General]
Hi, I'm looking into doing some work around retry topics auto-configuration with Spring Boot - in particular adding a few lines in the `SK` docs about it.

One concern I remember @garyrussell raised a while back was that auto creation of topics was disabled for [auto-configuration](https://github.com/spring-projects/spring-boot/blob/5817c8441d4a24ca0ad6efe70bec64d22875b652/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaAutoConfiguration.java#L175-L187):

```java
@Bean
@ConditionalOnProperty(name = "spring.kafka.retry.topic.enabled")
@ConditionalOnSingleCandidate(KafkaTemplate.class)
public RetryTopicConfiguration kafkaRetryTopicConfiguration(KafkaTemplate<?, ?> kafkaTemplate) {
    KafkaProperties.Retry.Topic retryTopic = this.properties.getRetry().getTopic();
    RetryTopicConfigurationBuilder builder = RetryTopicConfigurationBuilder.newInstance()
        .maxAttempts(retryTopic.getAttempts())
	.useSingleTopicForSameIntervals()
	.suffixTopicsWithIndexValues()
	.doNotAutoCreateRetryTopics();
    setBackOffPolicy(builder, retryTopic);
    return builder.create(kafkaTemplate);
}
```

I thought of adding an auto-configuration property for allowing users to override this, and opened an [issue in boot](https://github.com/spring-projects/spring-boot/issues/34046) for that.

But now that I think again, the default behavior for `KafkaConsumer` is to auto create topics, so I'm thinking this might be a non-issue? I just created a sample app and when using auto-configuration it creates all topics accordingly.

If this looks good enough, I'm thinking I'll just add a few lines to `SK` docs so users know they can use auto configuration for the feature.

How does that sound?

Thanks!

**Comment by garyrussell** [comment]
@tomazfernandes The problem with broker-auto-created topics is there is no control over the partitions/replication (at the application level). It can also be disabled at the broker level.

@snicoll reached out to me last week and he said he plans on addressing that Boot issue before the RC next week; we decided to go with your original suggestion (3 properties).

**Comment by tomazfernandes** [comment]
Cool, thanks @garyrussell! Let me know if there's anything I can help with.
